plugins/plugin-chart-echarts/src/Tree/controlPanel.tsx

```diff
@@ -0,0 +1,299 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+import React from 'react';
+import { t } from '@superset-ui/core';
+import { ControlPanelConfig, sections, sharedControls } from '@superset-ui/chart-controls';
+import { DEFAULT_FORM_DATA } from './types';
+
+const requiredEntity = {
+  ...sharedControls.entity,
+  clearable: false,
+};
+const optionalEntity = {
+  ...sharedControls.entity,
+  clearable: true,
+  validators: [],
+};
+
+const controlPanel: ControlPanelConfig = {
+  controlPanelSections: [
+    sections.legacyRegularTime,
+    {
+      label: t('Query'),
+      expanded: true,
+      controlSetRows: [
+        [
+          {
+            name: 'id',
+            config: {
+              ...requiredEntity,
+              label: t('Id'),
+              description: t('Name of the id column'),
+            },
+          },
+        ],
+        [
+          {
+            name: 'relation',
+            config: {
+              ...requiredEntity,
+              label: t('Relation'),
+              description: t('Name of the relation column'),
+            },
+          },
+        ],
+        [
+          {
+            name: 'name',
+            config: {
+              ...optionalEntity,
+              label: t('Name'),
+              description: t(
+                'Optional name of the data column.If not specified id will be used to name the nodes',
+              ),
+            },
+          },
+        ],
+        [
+          {
+            name: 'root_node',
+            config: {
+              ...requiredEntity,
+              type: 'TextControl',
+              label: t('Root node name'),
+              description: t('Name of root node of tree'),
+            },
+          },
+        ],
```

Comment:
I am unable to change the root node to any child node (the node appears alone without any children). Also, would it be possible to default to the first node with a null parent? That would make it more simple to get started.

Reply:
Weird there is actually a test for it.
If name col selected, this should be from that column else name from the id column should work.

Reply:
Ok, maybe I did something wrong! Btw, it may also be a good idea to add to the tooltip that this should be the id of the node if no name is selected; otherwise the name should be entered.

Reply:
Also how about always defaulting to the first node in response data as a parent of parent node may not be necessarily null,could be empty string,'/' in case of filesystem ?

Reply:
I think it would be neat if it could default to first checking which nodes don't have any parents, and once those have been found, it could then default to either the first one, or perhaps the one with most children.

Reply:
I will need some time to do this correctly,meanwhile this will check for null / 'null' parents to choose root node, if not found take the first entry from data.
Is it ok @villebro ?

Reply:
@villebro applied logic according to your suggestions,could you review it please?

Reply:
And only id is required to set root node and is independent on name column set or not.